geode-core/src/main/java/org/apache/geode/management/internal/api/LocatorClusterManagementService.java

```diff
@@ -0,0 +1,120 @@
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.geode.management.internal.api;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.cache.configuration.CacheConfig;
+import org.apache.geode.cache.configuration.ClusterCacheElement;
+import org.apache.geode.cache.execute.Function;
+import org.apache.geode.cache.execute.ResultCollector;
+import org.apache.geode.distributed.ConfigurationPersistenceService;
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.management.internal.cli.CliUtil;
+import org.apache.geode.management.internal.cli.functions.CliFunctionResult;
+import org.apache.geode.management.internal.cli.functions.UpdateCacheFunction;
+import org.apache.geode.management.internal.exceptions.EntityExistsException;
+import org.apache.geode.management.internal.exceptions.NoMembersException;
+
+public class LocatorClusterManagementService implements ClusterManagementService {
+  private static Logger logger = LogService.getLogger();
+  private InternalCache cache;
+  private ConfigurationPersistenceService persistenceService;
+
+  public LocatorClusterManagementService(InternalCache cache,
+      ConfigurationPersistenceService persistenceService) {
+    this.cache = cache;
+    this.persistenceService = persistenceService;
+  }
+
+  @Override
+  public APIResult createCacheElement(ClusterCacheElement config) {
+    APIResult result = new APIResult();
+    CacheConfig cacheConfig;
+    String group = "cluster";
+
+    // assuming config is validated at this point
+    if (persistenceService != null) {
+      cacheConfig = persistenceService.getCacheConfig(group, true);
+      // see if the element already exists in this group's configuration
+      if (config.getExisting(cacheConfig) != null) {
+        throw new EntityExistsException("cache element " + config.getId() + " already exists.");
+      }
+    }
+
+    // execute function on all members
+    Set<DistributedMember> targetedMembers = findMembers(null, null);
+    if (targetedMembers.size() == 0) {
+      throw new NoMembersException();
```

Comment:
This will eventually be handled by the REST API -- when it catches a NoMembersException, it knows exactly what error message to give. No need to introduce a special message.

Reply:
I disagree... The REST API should be "dumb" and only pass on the error that was given to it.. not make one up. :)

Reply:
I feel it might be useful to have the ability to throw different HTTP error codes based on the exception that happened. In this case, maybe it's just a 500 anyway, but in general if we make the REST API completely dumb, then it will be difficult to throw the right error code (outside of string-parsing the message).

Reply:
With "dumb" I meant, it is still ok to have the Exception, but with a usable message... maybe this message could even by localized.

Reply:
Sure, yeah we can add an error message, that can be bubbled down by the REST service.